assets/model_monitoring/components/src/model_data_collector_preprocessor/store_url.py

```diff
@@ -190,10 +190,15 @@ def _set_properties(self, ws: Workspace):
             self.path = matches.group("path").strip("/")
             self._datastore = None
         elif url.scheme == "azureml":
-            if ':' in url.path:  # azureml asset path
+            if ':' in url.path:  # azureml:<data_name>:<version> asset path
                 # asset path should be translated to azureml or hdfs path in service, should not reach here
                 raise InvalidInputError("AzureML asset path is not supported as uri_folder.")
-            else:  # azureml long or short form
+            elif '/data/' in self._base_url:  # azureml:// data asset url
```

Comment:
Is this a safe check we can make for distinguishing invalid paths? What if the data uri is in a folder called data?

Reply:
use regex to do the check